contracts/FeralfileArtworkV4.sol

```diff
@@ -76,37 +232,103 @@ contract FeralfileExhibitionV4 is
         return _contractURI;
     }
 
-    /// @notice isValidRequest validates a message by ecrecover to ensure
+    /// @notice verifies a message by ecrecover to ensure
     //          it is signed by owner of token.
     /// @param message_ - the raw message for signing
     /// @param owner_ - owner address of token
     /// @param r_ - part of signature for validating parameters integrity
     /// @param s_ - part of signature for validating parameters integrity
     /// @param v_ - part of signature for validating parameters integrity
-    function isValidRequest(
+    function verifySignature(
         bytes32 message_,
         address owner_,
         bytes32 r_,
         bytes32 s_,
         uint8 v_
     ) internal pure returns (bool) {
-        address signer = ECDSA.recover(
-            ECDSA.toEthSignedMessageHash(message_),
-            v_,
-            r_,
-            s_
+        return
+            ECDSA.recover(ECDSA.toEthSignedMessageHash(message_), v_, r_, s_) ==
+            owner_;
+    }
+
+    function _seriesExists(uint256 seriesId) internal view returns (bool) {
+        return _seriesMaxSupplies[seriesId] > 0;
+    }
```

Comment:
this logic could be a modifier so we could reuse it cleaner

Reply:
I do not think so!! if so, why not turn `_exists` into a modifier?

Reply:
I think it depends how to use it. modifier is quite strict that can only be run before executing the target function while a private function could be called flexibly.

Reply:
hmm I do some research and seems like the best practice is using the modifier so it could be reuse and override.
but actually using private function saving more gas while deploy the contract cause modifier copy paste the code into the function that use it while compiling.

Reply:
I'll change my mind and vot for private function :P